packages/commonwealth/client/scripts/views/components/collapsible_body_text.tsx

```diff
@@ -0,0 +1,122 @@
+/* @jsx m */
+
+import m from 'mithril';
+
+import app from 'state';
+import { Account, Comment, Thread, AnyProposal } from 'models';
+import { countLinesQuill, countLinesMarkdown } from './quill/helpers';
+import { MarkdownFormattedText } from './quill/markdown_formatted_text';
+import { QuillFormattedText } from './quill/quill_formatted_text';
+import User from './widgets/user';
+
+const QUILL_PROPOSAL_LINES_CUTOFF_LENGTH = 50;
+const MARKDOWN_PROPOSAL_LINES_CUTOFF_LENGTH = 70;
+
+export const formatBody = (vnode, updateCollapse) => {
+  const { item } = vnode.attrs;
+  if (!item) return;
+
+  const body =
+    item instanceof Comment
+      ? item.text
+      : item instanceof Thread
+      ? item.body
+      : item.description;
+  if (!body) return;
+
+  vnode.state.body = body;
+  if (updateCollapse) {
+    try {
+      const doc = JSON.parse(body);
+      if (countLinesQuill(doc.ops) > QUILL_PROPOSAL_LINES_CUTOFF_LENGTH) {
+        vnode.state.collapsed = true;
+      }
+    } catch (e) {
+      if (countLinesMarkdown(body) > MARKDOWN_PROPOSAL_LINES_CUTOFF_LENGTH) {
+        vnode.state.collapsed = true;
+      }
+    }
+  }
+};
+
+export class CollapsibleBodyText
+  implements
+    m.ClassComponent<{
+      item: AnyProposal | Thread | Comment<any>;
+    }>
+{
+  private body: any;
+  private collapsed: boolean;
+
+  oninit(vnode) {
+    this.collapsed = false;
+    formatBody(vnode, true);
+  }
+
+  onupdate(vnode) {
+    formatBody(vnode, false);
+  }
+
+  view(vnode) {
+    const { body } = this;
+
+    const getPlaceholder = () => {
+      if (!(vnode.attrs.item instanceof Thread)) return;
+
+      const author: Account = app.chain
+        ? app.chain.accounts.get(vnode.attrs.item.author)
+        : null;
+
+      return author ? (
+        <>
+          {m(User, {
+            user: author,
+            hideAvatar: true,
+            hideIdentityIcon: true,
+          })}{' '}
+          created this thread
+        </>
+      ) : (
+        'Created this thread'
+      );
+    };
+
+    const text = () => {
```

Comment:
i don't love the format of this component that just returns empty tags of running a function that returns various internal components. let's stick to our classic turnery or if/else tree structure for view() methods. this text() method is functioning like the view() but within the view() which is silly

Reply:
That code is grandfathered in from a long time ago. Safe to come back to it later?

Reply:
sounds good